File: Ser2023/Company/Company.Web/Modules/Company/Country/CountryRow.cs
using Company.Administration.Entities;
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System;
using System.ComponentModel;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[person].[Country]")]
[DisplayName("Country"), InstanceName("Country")]
[ReadPermission(PermissionKeys.Country.View)]
[ModifyPermission(PermissionKeys.Country.Modify)]
[DeletePermission(PermissionKeys.Country.Delete)]
[LookupScript("Company.Country", LookupType = typeof(MultiTenantRowLookupScript<>))]

public sealed class CountryRow : LoggingRow<CountryRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    [DisplayName("Country Code"), Size(3), PrimaryKey, NotNull, IdProperty, QuickSearch]
    public string CountryCode { get => fields.CountryCode[this]; set => fields.CountryCode[this] = value; }

    [DisplayName("Country Name"), Size(50), NotNull, NameProperty]
    public string CountryName { get => fields.CountryName[this]; set => fields.CountryName[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public StringField CountryCode;
        public StringField CountryName;

        public Int16Field IsActive;
        public Int32Field TenantId;
    }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/CountryRow.ts
import { fieldsProxy } from '@serenity-is/corelib';

export interface CountryRow {
    CountryCode?: string;
    CountryName?: string;
    IsActive?: number;
    TenantId?: number;
}

export abstract class CountryRow {
    static readonly idProperty = 'CountryCode';
    static readonly nameProperty = 'CountryCode';
    static readonly localTextPrefix = 'Company.Country';

    static readonly deletePermission = 'PermissionKeys.Country';
    static readonly insertPermission = 'PermissionKeys.Country';
    static readonly readPermission = 'PermissionKeys.Country';
    static readonly updatePermission = 'PermissionKeys.Country';

    static readonly Fields = fieldsProxy<CountryRow>();
}